src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1150,71 +1118,24 @@ public async Task GoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedExcep
 
-                    if (headerData.Contains("request1"))
-                    {
-                        done = true;
-                        HttpResponseMessage response1 = await sendTask1;
-                        Assert.Equal(HttpStatusCode.OK, response1.StatusCode);
-                    }
-                    else if (headerData.Contains("request2"))
-                    {
-                        Assert.False(request2Retried);
+                Assert.Contains("request1", headerData);
 
-                        (bool succeeded, HttpResponseMessage response2) = await IgnoreSpecificException<HttpRequestException, HttpResponseMessage>(sendTask2, "ENHANCE_YOUR_CALM");
-                        if (succeeded)
-                        {
-                            Assert.Equal(HttpStatusCode.OK, response2.StatusCode);
-                            request2Retried = true;
-                        }
-                    }
-                    else if (headerData.Contains("request3"))
-                    {
-                        Assert.False(request3Retried);
-
-                        (bool succeeded, HttpResponseMessage response3) = await IgnoreSpecificException<HttpRequestException, HttpResponseMessage>(sendTask3, "ENHANCE_YOUR_CALM");
-                        if (succeeded)
-                        {
-                            Assert.Equal(HttpStatusCode.OK, response3.StatusCode);
-                            request3Retried = true;
-                        }
-                    }
-                    else
-                    {
-                        Assert.True(false, "Unrecognized sender of the request");
-                    }
-
-                    lastRetriedStreamId = retriedStreamId;
-                }
+                HttpResponseMessage response1 = await sendTask1;
+                Assert.Equal(HttpStatusCode.OK, response1.StatusCode);
 
-                await newConnection.ShutdownIgnoringErrorsAsync(lastRetriedStreamId, ProtocolErrors.ENHANCE_YOUR_CALM);
+                await newConnection.ShutdownIgnoringErrorsAsync(retriedStreamId, ProtocolErrors.ENHANCE_YOUR_CALM);
                 await connection.WaitForConnectionShutdownAsync();
 
-                // We cannot tell for sure if request 2 or 3 already failed or got retried
-                // Now we make server stop listening to avoid any hangs due to retries and server still listening
-
-                server.Dispose();
-
-                if (!request2Retried)
-                {
-                    await AssertProtocolErrorAsync(sendTask2, ProtocolErrors.ENHANCE_YOUR_CALM, allowConnectionRefused: true);
-                }
-
-                if (!request3Retried)
-                {
-                    await AssertProtocolErrorAsync(sendTask3, ProtocolErrors.ENHANCE_YOUR_CALM, allowConnectionRefused: true);
-                }
+                await AssertProtocolErrorAsync(sendTask2, ProtocolErrors.ENHANCE_YOUR_CALM);
```

Comment:
Can we move this above the shutdown logic? This should happen even without waiting for shutdown